src/Cli/Microsoft.DotNet.Cli.Utils/TypoCorrection.cs

```diff
@@ -13,34 +13,97 @@ namespace Microsoft.DotNet.Cli.Utils
     public static class TypoCorrection
     {
         /// <summary>
-        /// Gets the list of tokens similar to <paramref name="currentToken"/>.
+        /// Gets the list of tokens similar to <paramref name="currentToken"/>
+        /// based on priority search:
+        /// 1. Starts with
+        /// 2. Contains - the call is restricted with <paramref name="currentToken"/> length check, minLength:3 and <param name="maxLevenshteinDistance">
+        /// 3. Levenshtein algorithm with <param name="maxLevenshteinDistance"> restriction
+        /// max number of suggestion is restricted to 10 entries
         /// </summary>
         /// <param name="possibleTokens">List of tokens to select from.</param>
         /// <param name="currentToken">The token that is being compared.</param>
         /// <param name="maxLevenshteinDistance">the difference between two strings, default: 3.</param>
         /// <returns>The enumerator to tokens similar to <paramref name="currentToken"/>.</returns>
         public static IEnumerable<string> GetSimilarTokens(IEnumerable<string> possibleTokens, string currentToken, int maxLevenshteinDistance = 3)
         {
-            int? bestDistance = null;
-            return possibleTokens
-                .Select(possibleMatch => (possibleMatch, distance: GetDistance(currentToken, possibleMatch)))
+            var minCurrentTokenLength = 3;
+            var maxNumberOfSuggestions = 10;
+
+            var currentTokenLength = currentToken.Length;
+            var startsWithAndContainsSpecificDistanceFunc = (string possibleMatch)
+                => (possibleMatch, distance: possibleMatch.Length - currentTokenLength);
+
+            var possibleMatchesTuples = possibleTokens
+                .Select(startsWithAndContainsSpecificDistanceFunc);
+
+            var matchByStartsWithTuples = possibleMatchesTuples
+               .Where(tuple => tuple.possibleMatch.StartsWith(currentToken))
+               .OrderBy(tuple => tuple.distance);
```

Comment:
it would be good to do ordering of results only once where possible

Reply:
I have applied your code changes. Please take a look